tornado/web.py

```diff
@@ -1286,17 +1286,21 @@ def get_browser_locale(self, default: str = "en_US") -> tornado.locale.Locale:
             locales = []
             for language in languages:
                 parts = language.strip().split(";")
-                if len(parts) > 1 and parts[1].strip().startswith("q="):
+                score = 1.0
+                for part in (
+                    x.split("=", 1)[1].strip()
+                    for x in parts[1:]
```

Comment:
This has gotten hard to follow with a generator comprehension inside a `for` loop expression (and the redundant split() calls). This would be cleaner if the loop were just `for part in parts[1:]` and then a `split()` call and `if` statement inside it.

Reply:
OK, I moved it into an `if` statement inside of the loop.